### PR TITLE
[pcl] Add cuda and tools option

### PR DESCRIPTION
**Pull request by cmpute**
With this portfile, pcl can be theoretically compiled against cuda. However, I met problems when building it with cuda like #2749. How I solved this is exactly described [here at StackOverflow](https://stackoverflow.com/questions/45918446/compiling-opencv-3-3-with-cuda-9-0rc) which was caused by CUDA cmake bug.

In addition, with current cuda version (8.0), only toolset along with VS2015 can be used to compile .cu files (while I am using VS2017). The problem is also describe in [another post at StackOverflow](https://stackoverflow.com/questions/42003849/nvcc-host-compiler-targets-unsupported-os), workarounds are mentioned in both posts. I am able to build pcl with cuda when specifying CUDA_HOST_COMPILER as the cl compiler in toolset v140. So the problem is with cuda rather than with pcl/opencv in my opinion, which could be solved or at least warned in the portfile of cuda.

### CODE REVIEW

**Comment by cmpute** [comment]
I found a way to workaround this, the codes are like following
```cmake
set(WITH_CUDA OFF)
if("cuda" IN_LIST FEATURES)
    set(WITH_CUDA ON)
    get_filename_component(CUDA_HOST_COMPILER "../../VC/bin/cl.exe" ABSOLUTE BASE_DIR "$ENV{VS140COMNTOOLS}"})
endif()

vcpkg_configure_cmake(... -DCUDA_HOST_COMPILER=${CUDA_HOST_COMPILER})
```
However, I found that `VS140COMNTOOLS` expanded to **nothing** when run from vcpkg, however, it expands normally to C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\ when use `cmake -P`. I have no idea how to deal with this, could you give me any suggestion? @ras0219-msft

File: ports/pcl/portfile.cmake [hunk 1]
```diff
@@ -36,23 +36,35 @@ if("pcap" IN_LIST FEATURES)
...
         -DPCL_BUILD_WITH_FLANN_DYNAMIC_LINKING_WIN32=${PCL_SHARED_LIBS}
         -DPCL_SHARED_LIBS=${PCL_SHARED_LIBS}
         # WITH
-        -DWITH_CUDA=OFF
+        -DWITH_CUDA=${WITH_CUDA}
         -DWITH_LIBUSB=OFF
         -DWITH_OPENNI2=${WITH_OPENNI2}
         -DWITH_PCAP=${WITH_PCAP}
-        -DWITH_PNG=OFF
+        -DWITH_PNG=ON
```
Comment (ras0219-msft) [inline ports/pcl/portfile.cmake]: Should `libpng` be added as a dependency because of this?

Comment (cmpute) [reply ports/pcl/portfile.cmake]: It could be, but `pcl` depends on `vtk` which takes `libpng` as a dependency, so adding it as dependency is not needed

**Comment by cmpute** [comment]
Now pcl can be built with cuda almost successfully. The problem occured during `vcpkg install pcl[cuda]` is described here at https://github.com/PointCloudLibrary/pcl/issues/2207.

**Comment by cmpute** [comment]
Is it able to merge this branch now?

**Comment by ras0219-msft** [comment]
Yep, thanks for the PR and sorry for the delay!

**Pull request merged**